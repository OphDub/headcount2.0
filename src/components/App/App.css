@import url("https://fonts.googleapis.com/css?family=Francois+One");

h1 {
    font-size: 45px;
    background: -webkit-linear-gradient(#eee, #333);
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
}

h1, form {
    text-align: center;
    font-family: 'Francois One', sans-serif;
}

form {
    padding: 20px;
    height: 30px;
}

input {
    font-size: 16px;
    text-align: center;
    height: 30px;
    width: 300px;
}