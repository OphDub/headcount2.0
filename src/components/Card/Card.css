article {
    border: 1px solid lightslategray;
    height: 325px;
    width: 200px;
    margin: 10px;
    font-family: 'Francois One', sans-serif;
}

ul {
    margin: auto;
    padding-left: 60px;
    list-style-type: none;
}

h4 {
    margin: auto;
    margin-bottom: 20px;
    padding: 10px;
    text-align: center;
    background-color: rgb(74, 169, 241);
    color: rgb(232, 238, 241);
}

.highFive {
    color: blue;
}

.lowFive {
    color: orangered;
}

.cardContainer article:hover {
    box-shadow: 4px 4px 0px 0px rgba(22, 22, 87, 0.2);
}

.compareContainer.selected {
  border: 3px solid rgb(57, 185, 231);
}

.selected {
    border: 3px solid rgb(57, 185, 231);
}