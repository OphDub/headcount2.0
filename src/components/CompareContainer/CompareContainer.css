.compareContainer article.Card {
    border: 10px solid blue;
}