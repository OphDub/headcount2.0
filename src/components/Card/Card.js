import React from 'react';
import { string, object } from 'prop-types';
import './Card.css';

const Card = ({ location, data }) => {
  // <article onClick={() => selectCard(location)}> ?
  const eachCard = Object.keys(data).map( dataPoint => {
                                            const highOrLow = data[ dataPoint ] >= 0.5 ? "highFive" : "lowFive";
                                            
                                            return <li className={ highOrLow } >{ dataPoint + ': ' + data[ dataPoint ] }</li>
                                        } )
  return (
    <article>
      <h4>{ location }</h4>
      <ul>{ eachCard }</ul>
    </article>
  )
}

Card.propTypes = {
  location: string.isRequired,
  data: object.isRequired,
}

export default Card;
