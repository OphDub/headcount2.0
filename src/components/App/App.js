import React, { Component } from 'react';
import './App.css';
import Search from '../Search/Search';
import CardContainer from '../CardContainer/CardContainer';
import CompareContainer from '../CompareContainer/CompareContainer';
import DistrictRepository from '../../helper';
import kinderData from '../../data/kindergartners_in_full_day_program';
import { fdatasync } from 'fs';
const masterDistrict = new DistrictRepository (kinderData)

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      allDistricts: [],
      comparedDistricts: [],
      comparisonObj: {}
    }
  }

  componentDidMount() {
    const kinderDistrict = masterDistrict.findAllMatches();
    this.setState({ allDistricts: kinderDistrict });
  }

  filterDistricts = (string) => {
    const upcaseString = string.toUpperCase();
    const foundDistricts = masterDistrict.findAllMatches(upcaseString);

    this.setState({ allDistricts: foundDistricts });
  }

  selectCard = (id) => {
    const foundDistrict = this.state.allDistricts.find((district)=> district.location === id)
    const newFoundDistrictArray = [...this.state.comparedDistricts, foundDistrict]
    const stateLocations = this.state.comparedDistricts.map( district => district.location)

    if (!stateLocations.includes(foundDistrict.location) && newFoundDistrictArray.length <= 2 ) {
      this.setState({ comparedDistricts: newFoundDistrictArray })

      this.compareCardData(newFoundDistrictArray);
    } else {
      const oldArray = this.state.comparedDistricts

      oldArray.shift()

      const newArray = [...oldArray, foundDistrict]

      this.setState({ comparedDistricts: newArray });
    }

    this.addCss(id)
  }

  compareCardData = (newFoundDistrictArray) => {
    if(newFoundDistrictArray.length === 2) {
      const comparisonObj = masterDistrict.compareDistrictAverages(newFoundDistrictArray[0].location, 
                                                                 newFoundDistrictArray[1].location);
      this.setState({ comparisonObj })
    }
  }

  addCss = (id) => {
    if(document.getElementById(id).className === "unselected" && document.getElementById(id).parentElement.className !== 'compareContainer') {
      document.getElementById(id).setAttribute("class", "selected")

    } else if (document.getElementById(id).className === "selected" && document.getElementById(id).parentElement.className === 'cardContainer'){
      document.getElementById(id).setAttribute("class", "unselected")
    }
  }

  render() {
    return (
      <div>
        <h1>Welcome To Headcount 2.0</h1>
        <Search filterDistricts={this.filterDistricts} />
        <CompareContainer comparedDistricts={this.state.comparedDistricts} 
                          selectCard={this.selectCard} 
                          comparisonObj={this.state.comparisonObj} />
        <CardContainer districts={this.state.allDistricts} selectCard={this.selectCard} />
      </div>
    );
  }
}

export default App;
