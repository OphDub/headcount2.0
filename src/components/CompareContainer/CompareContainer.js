import React from 'react';
import './CompareContainer.css';
import Card from '../Card/Card';

const CompareContainer = ({ comparedDistricts, selectCard, comparisonObj }) => {
  const averageComparison = comparisonObj.compared ? "Combined Average:  " + comparisonObj.compared : null
  const comparisonArr = Object.entries(comparisonObj).map( (entry) => <div className="displayData">{entry[0]} <br/> <span>Average Enrollment:  {entry[1]}</span></div> )
  const renderedComparison = comparedDistricts.map( district =>
                                                            <Card {...district}
                                                                  key={district.location}
                                                                  addCss={'selected'}
                                                                  id={district.location}
                                                                  selectCard={selectCard}/>)

  return (
    <section className="compareContainer">
      {renderedComparison[0]}
      {renderedComparison.length > 0 &&
        <article className="comparisonCard">
          <div>{comparisonArr[0]}</div>
          <div>{comparisonObj.compared}</div>
          <div>{comparisonArr[1]}</div>
        </article>
      }
      {renderedComparison[1]}
    </section>
  )
}

export default CompareContainer;
