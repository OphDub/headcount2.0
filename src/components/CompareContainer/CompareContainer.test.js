import React from 'react';
import CompareContainer from './CompareContainer';
import Card from '../Card/Card';
import App from '../App/App';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DistrictRepository from '../../helper';
import kinderData from '../../data/kindergartners_in_full_day_program';

describe('COMPARE CONTAINER', () => {
  let wrapper;
  let mockSelectFunc;
  let comparedDistrictsArr;
  let comparisonObj;
  let renderedComponent;
  let masterDistrict;

  beforeEach(() => {
    renderedComponent = shallow(<App />);
    masterDistrict = new DistrictRepository(kinderData);
    mockSelectFunc = jest.fn();
    comparisonObj = {'CAMPO RE-6': 0.657, 'CANON CITY RE-1': 0.709, compared: 0.927};
    comparedDistrictsArr = [{ location: 'CAMPO RE-6',
      data: 
                        { '2004': 0,
                          '2005': 0,
                          '2006': 1,
                          '2007': 1,
                          '2008': 1,
                          '2009': 1,
                          '2010': 1,
                          '2011': 1,
                          '2012': 1,
                          '2013': 1,
                          '2014': 1 } },
    { location: 'CANON CITY RE-1',
      data:
                        { '2004': 0.302,
                          '2005': 0.353,
                          '2006': 0.119,
                          '2007': 0.218,
                          '2008': 0.766,
                          '2009': 0.98,
                          '2010': 0.993,
                          '2011': 0.993,
                          '2012': 1,
                          '2013': 1,
                          '2014': 1 } }];
    wrapper = shallow(<CompareContainer comparedDistricts={comparedDistrictsArr} selectCard={mockSelectFunc} comparisonObj={comparisonObj} />);
  });

  it('it should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('it should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain the correct number of Cards', () => {
    expect(wrapper.find(Card).length).toEqual(2);
  });

  it('should render compare cards correctly', () => {
    const newComparisonObj = masterDistrict.compareDistrictAverages(comparedDistrictsArr[0].location, 
      comparedDistrictsArr[1].location);
    renderedComponent.setState({ newComparisonObj });
    
    expect(wrapper).toMatchSnapshot();
  });
});
