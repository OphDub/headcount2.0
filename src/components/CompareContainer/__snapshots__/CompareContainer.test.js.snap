// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPARE CONTAINER it should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Compare Container",
    },
    "ref": null,
    "rendered": "Compare Container",
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Compare Container",
      },
      "ref": null,
      "rendered": "Compare Container",
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
